VTS/Core/Implementations/WebSocketNetCore/WebSocketNetCoreImpl.cs

```diff
@@ -0,0 +1,213 @@
+                {
+                    _onConnect();
+                    _logger.Log($"[{_url}] - Socket open!");
+                    _attemptReconnect = true;
+                });
+
+                while (true)
+                {
+                    var result = await _socket.ReceiveAsync(CancellationToken.None);
+                    if (result.closeStatus == null)
+                    {
+                        _responseQueue.Enqueue(() =>
+                        {
+                            if (result is { buffer: { }, messageType: WebSocketMessageType.Text })
+                            {
+                                _intakeQueue.Enqueue(Encoder.GetString(result.buffer));
+                            }
+                        });
+                    }
+                    else
+                    {
+                        _responseQueue.Enqueue(() =>
+                        {
+                            var msg =
+                                $"[{_url}] - Socket closing: {result.closeStatus}, '{result.closeStatusDescription}', {result.closeStatus == WebSocketCloseStatus.NormalClosure}";
+                            if (result.closeStatus == WebSocketCloseStatus.NormalClosure)
+                            {
+                                _logger.Log(msg);
+                                _onDisconnect();
+                            }
+                            else
+                            {
+                                _logger.LogError(msg);
+                                _onError(new Exception(msg));
+                                if (_attemptReconnect)
+                                {
+                                    Reconnect();
+                                }
+                            }
+                        });
+                    }
+                }
+            }, CancellationToken.None);
+        }
+
+        public void Stop()
+        {
+            _attemptReconnect = false;
+            if (_socket is { State: WebSocketState.Open })
+            {
+                _socket.Abort();
+            }
+        }
+
+        private void Reconnect()
+        {
+            Start(_url, _onConnect, _onDisconnect, _onError);
+        }
+
+        public void Tick(float timeDelta)
+        {
+            do
+            {
+                if (_responseQueue.IsEmpty || !_responseQueue.TryDequeue(out var action))
+                    continue;
+
+                try
+                {
+                    action();
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError($"Socket error: {e.StackTrace}");
+                }
+            } while (!_responseQueue.IsEmpty);
+        }
+    }
+}
+
+
+internal static class WebSocketExtensions
+{
+    public static async Task<(
+        byte[] buffer,
+        WebSocketMessageType messageType,
+        WebSocketCloseStatus? closeStatus,
+        string? closeStatusDescription
```

Comment:
Same as earlier, Nullable Properties do not appear to be supported in Unity. Is there a reason that this needs to be marked as Nullable? `strings` can already be null, as they're not structs.

Reply:
In .NET 6+ projects you can enable strict nullable checking. Even `string`s need to be explicitly set to be nullable.
Helps in keeping unintentional errors in check.